docker/check_production_base_image.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# First check that the Docker image exists on the local machine
+if [[ ${cardsBaseImage} == cards/cards:latest ]]
+then
+  docker image inspect ${cardsBaseImage} >/dev/null 2>/dev/null
+  if [[ $? -ne 0 ]]
+  then
+    echo "No local image found, please rebuild it following the instructions for building a production-ready docker image in the CARDS README, then try again." 1>&2
+    exit -1
+  fi
+fi
+
```

Comment:
I think it would also be good to check that the local `cards/cards:latest` image is for the _SNAPSHOT_ version of CARDS specified by the specific project. For example, if I run `setup.sh` and specify `0.9.24-SNAPSHOT` as the CARDS version as that is what `cards/cards:latest` is when `setup.sh` was ran but then `cards/cards:latest` is replaced with a new image of the generic CARDS platform but this time of version `0.9.25-SNAPSHOT` this build should fail.